arwen/contexts/runtime.go

```diff
@@ -188,7 +189,43 @@ func (context *runtimeContext) GetVMInput() *vmcommon.VMInput {
 }
 
 func (context *runtimeContext) SetVMInput(vmInput *vmcommon.VMInput) {
-	context.vmInput = vmInput
+	if !context.host.IsArwenV2Enabled() || vmInput == nil {
+		context.vmInput = vmInput
+		return
+	}
+
+	context.vmInput = &vmcommon.VMInput{
+		CallType:    vmInput.CallType,
+		GasPrice:    vmInput.GasPrice,
+		GasProvided: vmInput.GasProvided,
```

Comment:
Should we also initialise `CallValue` with 0?

Reply:
it is set below

Reply:
done